GoMeet/src/test/java/com/google/sps/LocationServletTest.java

```diff
@@ -90,4 +101,30 @@ public void doPostTest1() {
   public void doPostTest2() {
     doPostTest();
   }
+
+  /** 
+   * Tests if stored locations are returned as a JSON string when there is 
+   * one location stored.
+   */
+  @Test
+  public void doGetTest() throws IOException {
+    // Set up database
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+    Entity location = new Entity("Location");
+    location.setProperty("title", LOCATION_A.getTitle());
+    location.setProperty("lat", LOCATION_A.getLat());
+    location.setProperty("lng", LOCATION_A.getLng());
+    location.setProperty("note", LOCATION_A.getNote());
+    ds.put(location);
+
+    Gson gson = new Gson();
+    String expectedJson = gson.toJson(new ArrayList<>(Arrays.asList(LOCATION_A)));
+
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    when(response.getWriter()).thenReturn(writer);
+
+    new LocationServlet().doGet(request, response);
+    assertTrue(stringWriter.toString().contains(expectedJson));
+  }
```

Comment:
I think this test relies too much on the implementation of `doGet()`, specifically that it uses Gson. JSON can be formatted in multiple ways with whitespace in between - e.g. `["Hello", "World"]` is the same as
```
[
    "Hello",
    "World"
]
```

And this test relies that the function's output is the same as Gson's JSON string output. Suppose you changed the JSON library in the function, and the new library adds whitespace in different locations - the code would still be correct, but the test would not pass.

What I think you want to test here is that doGet() outputs valid JSON and that the JSON contains an object you specify - so you should instead parse the output string with Gson, and then check that the object contains the expected properties.